### ISSUE TITLE
Stale adding documents if there are too many ongoing merge 

### ISSUE THREAD

**Issue by fulmicoton** [quickwit-oss/tantivy#680]
Following #677 we should have a way to stale indexing if there are too many ongoing merge threads.

**Comment by vsop-479** [comment]
@fulmicoton 
It seems tantivy has fixed this bug by using ThreadPoolBuilder for merge threads.
If so, I think it would be good to close this issue.

**Comment by fulmicoton** [comment]
That's incorrect.  We avoid spawning too many merges, but we do not prevent people from committing.
